src/osd/PG.cc

```diff
@@ -4182,9 +4182,10 @@ void PG::_scan_snaps(ScrubMap &smap)
 			    << "...repaired";
 	}
 	snap_mapper.add_oid(hoid, obj_snaps, &_t);
-	r = osd->store->apply_transaction(osr.get(), std::move(t));
+	// we don't particularly care when this is visible or committed.
+	r = osd->store->queue_transaction(osr.get(), std::move(t), nullptr);
```

Comment:
Do we need to worry about skipping a snaptrim, because it gets processed before the repair is complete?